File: dev/test_data/golang/test_main/aas_core_meta.v3/input/snippets/Verification/data_specification_IEC_61360s_have_data_type.go
// Check that [aastypes.DataSpecificationIec61360.DataType]
// is defined for all data specifications whose content is given as
// IEC 61360.
func DataSpecificationIEC61360sHaveDataType(
	embeddedDataSpecifications []aastypes.IEmbeddedDataSpecification) bool {
	for _, eds := range embeddedDataSpecifications {
		content := eds.DataSpecificationContent()

		ok := aastypes.IsDataSpecificationIEC61360(content)
		if !ok {
			continue
		}
		iec61360 := content.(aastypes.IDataSpecificationIEC61360)

		dt := iec61360.DataType()
		if dt == nil {
			return false
		}
	}
	return true
}

File: dev/test_data/test_main/csharp/aas_core_meta.v3/input/snippets/Verification/data_specification_IEC_61360s_have_data_type.cs
/// <summary>
/// Check that the <see cref="Aas.IDataSpecificationIec61360.DataType" /> is defined
/// for all data specifications whose content is given as IEC 61360.
/// </summary>
[CodeAnalysis.SuppressMessage("ReSharper", "InconsistentNaming")]
public static bool DataSpecificationIec61360sHaveDataType(
    IEnumerable<Aas.IEmbeddedDataSpecification> embeddedDataSpecifications
)
{
    foreach (var embeddedDataSpecification in embeddedDataSpecifications)
    {
        var iec61360 = (
            embeddedDataSpecification.DataSpecificationContent
                as Aas.IDataSpecificationIec61360
        );
        if (iec61360 != null)
        {
            if (iec61360.DataType == null)
            {
                return false;
            }
        }
    }

    return true;
}
